src/Serializer/AbstractCollectionNormalizer.php

```diff
@@ -101,7 +101,7 @@ public function normalize(mixed $object, ?string $format = null, array $context
      */
     protected function normalizeRawCollection(iterable $object, ?string $format = null, array $context = []): array|\ArrayObject
     {
-        if (!$object && ($context[Serializer::EMPTY_ARRAY_AS_OBJECT] ?? false) && \is_array($object)) {
+        if ([] === $object && ($context[Serializer::EMPTY_ARRAY_AS_OBJECT] ?? false)) {
```

Comment:
!$object already enforce it's `[]`.

But I rewrote the condition to be explicit